ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/VulnReportsPage.tsx

```diff
@@ -190,6 +270,43 @@ function VulnReportsPage() {
                     </Card>
                 </PageSection>
             </PageSection>
+            {reportIdToDelete !== '' && (
+                <Modal
+                    variant="small"
+                    title="Permanently delete report?"
+                    isOpen={isModalOpen}
+                    onClose={closeDeleteModal}
+                    actions={[
+                        <Button
+                            key="confirm"
+                            variant="danger"
+                            isLoading={isDeleting}
+                            onClick={() => {
+                                // eslint-disable-next-line no-void
+                                void deleteReport(reportIdToDelete);
```

Comment:
So I couldn't really find a good way to get around this issue. If I just did `deleteReport(reportIdToDelete)` then I get the following typescript error `Promises must be handled appropriately or explicitly marked as ignored with the `void` operator`. But adding a `void` just causes another typescript error `Expected 'undefined' and instead saw 'void'.`. So I added this disable. Has anyone ever encountered this? (cc: @pedrottimark @dvail @bradr5 @vjwilson @alwayshooin )

Reply:
Since `deleteReport` returns a Promise our linter rules want you to handle the error case in some way. You could instead do
```typescript
deleteReport(reportIdToDelete).catch((err) => {
     // Do something (nothing?)
});
```
but I'm not sure what a valid way to handle this would be - the only logical errors could come from the `catch` block in the defined function.

You could also do
```
// eslint-disable-next-line @typescript-eslint/no-floating-promises
deleteReport(reportIdToDelete);
```
to suppress it without the `void`.

Functionally I think any of the three methods to just ignore it would work, even if none of them are satisfying.

Reply:
@dvail Thanks, those are good suggestions. I was thinking of handling the promise using a `then` and `catch`, but then it did feel odd because I'm basically handling the promise returned from the service function within the `deleteReport` function too (just in the async/await format). My other thought was to take the logic within the `deleteReport` function and just put that in the `onClick` callback function. That would probably work but I usually prefer defining callback functions further up in the component and passing them down to components.

@pedrottimark Thanks, that's pretty neat. I think that works for my case, so I'll try that out!